src/http/routes.cc

```diff
@@ -140,6 +140,11 @@ routes& routes::add(operation_type type, const url& url,
     return add(rule, type);
 }
 
+routes& routes::add_default_handler(handler_base* handler) {
+    _default_handler = handler;
```

Comment:
who will delete this handler?

Reply:
The calling application, as the application can choose to pass an object from the stack as the handler, I think this responsibility should not be seastar's.

Reply:
ok, just make sure you run it in debug mode and see that you don't get errors in the shutdown process, it seems like a potential problem

Reply:
Have validated locally, good to go from my side.

Reply:
@amnonh this is a pre-existing "feature" in `add()` and `put()` as well - the caller needs to "new" a handler object, pass it by pointer, and it will never get freed. This is indeed very un-C++-like. Something like std::function<> would have probably made more sense.  But it makes sense for the new function to take the same "handler" type as the existing functions.